fe/fe-core/src/main/java/com/starrocks/common/util/concurrent/CountingLatch.java

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.common.util.concurrent;
+
+import com.google.common.base.Preconditions;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * CountDownLatch with extra counting up functionality.
+ */
+public class CountingLatch {
```

Comment:
You can move the name to the `ConditionVariable`? 
As well, I think you can use the lock and condition to implement a `ConditionVariable`
```
public class ConditionVariable {
    private final Lock lock = new ReentrantLock();
    private final Condition condition = lock.newCondition();
    private boolean conditionMet = false;

    public void await() throws InterruptedException {
        lock.lock();
        try {
            while (!conditionMet) {
                condition.await();
            }
        } finally {
            lock.unlock();
        }
    }
  
    public void await(long time, TimeUnit unit) throws InterruptedException {
        lock.lock();
        try {
            if (!conditionMet) {
                condition.await(time, unit);
            }
        } finally {
            lock.unlock();
        }
    }

    public void signal() {
        lock.lock();
        try {
            conditionMet = true;
            condition.signal();
        } finally {
            lock.unlock();
        }
    }
}
```

Reply:
1. CountingLatch is more general than conditional var, i have another pr which has dependency on this class.
2. And most importantly, your `ConditionVariable` implementation cannot apply in this PR, because all the concurrent threads will stuck on `await`, and no one will wake them up?
3. Plz don't copy and paste the code from `chatGPT`...Have you ever thought about how to use your code for this PR?